File: notification-service/src/main/java/com/Man10h/notification_service/model/entities/Notification.java
package com.Man10h.notification_service.model.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Date;

@Entity
@Table(name = "notification")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Notification {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String userId;
    private String content;
    private Boolean isRead;
    private LocalDateTime createdAt;
    private String targetCode;
}

File: frontend/src/types/notification.ts
export interface NotificationItem {
  id: number;
  userId: string;
  content: string;
  isRead: boolean;
  createdAt: string; // ISO LocalDateTime format, e.g. "2026-07-23T21:19:00"
  targetCode: string;
}

export interface NotificationPageResponse {
  content: NotificationItem[];
  totalElements: number;
  totalPages: number;
  page: number;
  size: number;
}
